modules/karpenter-provisioners/variables.tf

```diff
@@ -0,0 +1,445 @@
+variable "karpenter_node_pools" {
+  type = list(object({
+    name                      = string
+    instance_types            = list(string)
+    kubernetes_labels         = map(string)
+    annotations               = optional(map(string), null)
+    ttl_seconds_after_empty   = optional(number, 300)
+    ttl_seconds_until_expired = optional(number, 2592000)
+    weight                    = optional(number, 0)
+    limits                    = optional(map(any), null)
+
+    taints = optional(
+      list(object(
+        {
+          key    = string
+          effect = string
+        }
+    )), null)
+
+    startup_taints = optional(
+      list(object(
+        {
+          key    = string
+          effect = string
+        }
+    )), null)
+
+    requirements = optional(
+      list(object(
+        {
+          key      = string
+          operator = string
+          values   = list(string)
+        }
+        )), [
+        {
+          key      = "karpenter.sh/capacity-type"
```

Comment:
keep only `karpenter.sh/capacity-type` and `kubernetes.io/arch` and remove the rest of things

Reply:
Updated